Plandomizer.py

```diff
@@ -1063,18 +1073,37 @@ def is_pattern(pattern):
     return pattern.startswith('!') or pattern.startswith('*') or pattern.startswith('#') or pattern.endswith('*')
 
 
-def pattern_matcher(pattern):
+def pattern_matcher(self, pattern):
```

Comment:
Here's what I'd propose instead: generate this group list in WorldDistribution on init, move pattern_matcher into WorldDistribution, and then if pattern is `#MajorItem` we have the group already built. (This should save list modification on every call to `pattern_matcher`.)

Reply:
Of course, as discussed in Discord, the other alternative is to change the predicate passed to ItemIterator in `pool_add_item` and `pool_replace_item` to check `item.majoritem` (and potentially another property specifically to avoid placing items like GF when it's in a fixed place, since `.majoritem` doesn't cover all of this).